shared/js/background/storage/https.es6.js

```diff
@@ -0,0 +1,61 @@
+const load = require('./../load.es6')
+const Dexie = require('dexie')
+const constants = require('../../../data/constants')
+
+class HTTPSStorage {
+    constructor () {
+        this.dbc = new Dexie('https')
+        this.dbc.version(1).stores({
+            httpsStorage: 'name,type,data,checksum'
+        })
+    }
+
+    // Load https data defined in constants.httpsLists.
+    // First try to grab new data via xhr. If that fails
+    // fall back to local db. 
+    getLists () {
+        return Promise.all(constants.httpsLists.map(list => {
+            return new Promise((resolve, reject) => {
+                
+                this.getDataXHR(list.url).then(data => {
+                    if (data) {
+                        // if we have new data store it in local DB for later
+                        this.storeInLocalDB(list.name, list.type, data)
+                        list.data = data
+                        resolve(list)
+                    } else {
+                        // No new data, look up old data from DB
+                        this.getDataFromLocalDB(list.name).then(storedData => {
+                            if (storedData && storedData.data) {
+                                list.data = storedData.data
+                                resolve(list)
+                            } else {
+                                reject(new Error(`HTTPS: no stored data for ${list.name}`))
+                            }
+                        })
+                    }
+                })
+            })
+        }))
+    }
+
+    getDataXHR (url) {
+        return new Promise((resolve, reject) => {
+            load.JSONfromExternalFile(url, resolve)
+            // if load fails, resolve and try to load from db instead
+            setTimeout(() => resolve(), 30000)
```

Comment:
Wait 30sec in cases where an xhr fails. The promise resolves without any data and falls back to the database here https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/235/files#diff-b3d1f859c76ee3ad6e55e5ccd6500bbaR28